containers-workarounds.md

```diff
@@ -11,13 +11,98 @@ We've documented ways to leverage these products below:
 | :-----                    |:-----                         | :-----                 | :-----          |
 | ArgoCD | https://github.com/argoproj/argo-cd/issues/3956 | build arm64 images from source / existing arm64 binaries | |
 | Flatcar Linux | https://github.com/kinvolk/Flatcar/issues/97 | arm64 support for alpha/edge | |
+| Istio | | [Build instruction below](#Istio) | |
 
+### Istio
 
-			
+Istio doesn't publish arm64 containers, but you can build them yourself by following the steps below.
 
+0. Launch a Graviton instance. These instructions were created using Ubuntu 20.04 (Focal), however, they should work with some modifications to the first step on other distributions.
 
+1. Install Docker and Golang. Use Docker's own package repository because, at the time of writing, the build depends on a feature in `buildx` which is not present in Ubuntu's packaged version. The code below can be copied and pasted in its entirety onto the shell. For other distributions, refer to [Docker's documentation](https://docs.docker.com/engine/install/).
+```
+sudo apt-get update && sudo apt-get upgrade -y && sudo apt-get install -y \
+     apt-transport-https \
+     ca-certificates \
+     curl \
+     gnupg \
+     lsb-release \
+     build-essential &&
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg &&
+echo \
+   "deb [arch=arm64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+   $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null &&
+sudo apt-get update && sudo apt-get install -y docker-ce docker-ce-cli containerd.io &&
+sudo systemctl enable docker && sudo systemctl start docker && sudo usermod -a -G docker ubuntu &&
+wget https://golang.org/dl/go1.16.5.linux-arm64.tar.gz &&
+sudo tar -C /usr/local/ -xzf go1.16.5.linux-arm64.tar.gz &&
+sudo ln -s /usr/local/go/bin/go /usr/local/bin/go
+```
 
+2. Logout and log back in after the previous step. Optionally reboot to apply any kernel updates that were downloaded.
 
+3. Clone the repositories.
+```
+cd ~ &&
+mkdir istio && cd istio &&
+git clone https://github.com/istio/istio.git &&
+git clone https://github.com/istio/proxy.git &&
+git clone https://github.com/istio/tools.git
+```
 
+3.1. Optionally checkout the exact versions tested for this guide. If you run into problems, try this first. (Note: these commands will reset the master branch, so do not execute them if you have your own unpublished work on the master branch.)
+```
+pushd istio && git checkout -B master 49dfd655557fa5e1f678d6c18761603ceddcb1f8 && popd &&
```

Comment:
are these random commits or specific tagged commits? are there specific released tags we can specify instead (e.g. the last release)?

Reply:
They just the ones I tested. I'll check to see if there is a recent release that has all of the necessary changes in it.

Reply:
It looks like the tag `1.11.0-rc.2` is sufficiently new enough for the proxy and tools repos, but there isn't a new enough tag on the istio repo that includes the necessary changes. I'll have to test again to be sure that it still works when checking out the tagged version, so I'll do that before I update it.

Reply:
I started a test and unfortunately when you checkout something other than the master branch, some script somewhere recognizes that and tags the docker image with the git tag name. That messes up the rest of the build script for istio. 😞

I think I'd like to merge it as is and I can figure out how to out smart that tagging logic afterward, if that's okay with you.

Reply:
ok. Can you address my other comment?

Reply:
Ah, sorry. I made the change, but I must have forgotten to push it. I've done that now.